From 3201db3835c85a1d9a407e621bff7902896e8a82 Mon Sep 17 00:00:00 2001
From: Alexander Aring <aahringo@redhat.com>
Date: Tue, 7 Sep 2021 12:06:00 -0400
Subject: [PATCH 06/14] dlm_controld: add version check for libquorum

This patch adds a simple version check for libquorum. Since commit
2e893b98 ("dlm_controld: use new quorum api to detect missed failures")
dlm_controld uses functionality which is only available in libquorum
3.1.0 and upwards.
---
 dlm_controld/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dlm_controld/Makefile b/dlm_controld/Makefile
index 667998076acc..8cfc97e6909a 100644
--- a/dlm_controld/Makefile
+++ b/dlm_controld/Makefile
@@ -60,6 +60,10 @@ ifeq ($(USE_SD_NOTIFY),yes)
 	BIN_LDFLAGS += $(shell $(PKG_CONFIG) --libs libsystemd)
 endif
 
+ifeq (, $(shell $(PKG_CONFIG) --libs "libquorum >= 3.1.0"))
+	 $(error "Requires libquorum at least version 3.1.0")
+endif
+
 all: $(LIB_TARGET) $(BIN_TARGET) $(LIB_PC)
 
 $(BIN_TARGET): $(BIN_SOURCE)
-- 
2.33.0

